lexicon/__main__.py

```diff
@@ -60,17 +64,64 @@ def MainParser():
 
     return parser
 
-#dynamically determine all the providers available.
+# Print returned JSON into a nice table for command line usage
+def print_json_result(logger, output=None, without_header=None):
+    try:
+        _ = (entry for entry in output)
+    except:
+        logger.debug('Command output is not iterable, and then cannot be printed with --quiet parameter not enabled.')
+        return
+
+    array = [[row['id'], row['type'], row['name'], row['content'], row['ttl']] for row in output]
+
+    # Insert header (insert before calculating the max width of each column to take headers size into account)
+    if not without_header:
+        headers = ['ID', 'TYPE', 'NAME', 'CONTENT', 'TTL']
+        array.insert(0, headers)
+
+    columnWidths = [0, 0, 0, 0, 0]
+    # Find max width for each column
+    for row in array:
+        for idx, col in enumerate(row):
+            width = len(str(col))
+            if width > columnWidths[idx]:
+                columnWidths[idx] = width
+
+    # Add a "nice" separator
+    if not without_header:
+        array.insert(1, ['-' * columnWidths[idx] for idx in range(len(columnWidths))])
+
+    # Construct table to be printed
+    print_table = []
+    for row in array:
+        rowList = []
+        for idx, col in enumerate(row):
+            rowList.append(str(col).ljust(columnWidths[idx]))
+        print_table.append(' '.join(rowList))
+
+    # Print table
+    print('\n'.join(print_table))
+
+# Dynamically determine all the providers available.
 def main():
-
     parsed_args = MainParser().parse_args()
     log_level = logging.getLevelName(parsed_args.log_level)
     logging.basicConfig(stream=sys.stdout, level=log_level, format='%(message)s')
 
     logger.debug('Arguments: %s', parsed_args)
     client = Client(parsed_args.__dict__)
-    client.execute()
-
+    
+    output = client.execute()
+
+    if not parsed_args.__dict__['quiet']:
```

Comment:
Just curious - why not `parsed_args.quiet`?

Reply:
Juste to be consistent with how parsed arguments are passed in the existing code (as a dict).

Reply:
Yes, consistency - just look few lines above, here the `parsed_args` is used differently:

```
log_level = logging.getLevelName(parsed_args.log_level)
```

Reply:
Right, I was more focused about the way parsed_args is passed to the provider, as `parsed_args.__dict__`

Reply:
And new commit for that.